pkg/oc/admin/migrate/legacyhpa/hpa.go

```diff
@@ -0,0 +1,209 @@
+	autoscaling "k8s.io/kubernetes/pkg/apis/autoscaling"
+	autoscalingclient "k8s.io/kubernetes/pkg/client/clientset_generated/internalclientset/typed/autoscaling/internalversion"
+	"k8s.io/kubernetes/pkg/kubectl/cmd/templates"
+	kcmdutil "k8s.io/kubernetes/pkg/kubectl/cmd/util"
+	"k8s.io/kubernetes/pkg/kubectl/resource"
+
+	"github.com/openshift/origin/pkg/oc/admin/migrate"
+	"github.com/openshift/origin/pkg/oc/cli/util/clientcmd"
+)
+
+var (
+	defaultMigrations = map[versionKind]versionKind{
+		// legacy oapi group
+		{"v1", "DeploymentConfig"}: {"apps.openshift.io/v1", "DeploymentConfig"},
+
+		// webconsole shenaniganry
+		{"extensions/v1beta1", "DeploymentConfig"}:      {"apps.openshift.io/v1", "DeploymentConfig"},
+		{"extensions/v1beta1", "Deployment"}:            {"apps/v1", "Deployment"},
+		{"extensions/v1beta1", "ReplicaSet"}:            {"apps/v1", "ReplicaSet"},
+		{"extensions/v1beta1", "ReplicationController"}: {"v1", "ReplicationController"},
+	}
+
+	internalMigrateLegacyHPALong = templates.LongDesc(fmt.Sprintf(`
+		Migrate Horizontal Pod Autoscalers to refer to new API groups
+
+		This command locates and updates every Horizontal Pod Autoscaler which refers to a particular
+		group-version-kind to refer to some other, equivalent group-version-kind.
+
+		The following transformations will occur:
+
+%s`, prettyPrintMigrations(defaultMigrations)))
+
+	internalMigrateLegacyHPAExample = templates.Examples(`
+		# Perform a dry-run of updating all objects
+	  %[1]s
+
+	  # To actually perform the update, the confirm flag must be appended
+	  %[1]s --confirm
+
+	  # Migrate a specific group-version-kind to the latest preferred version
+	  %[1]s --initial=extensions/v1beta1.ReplicaSet --confirm
+
+	  # Migrate a specific group-version-kind to a specific group-version-kind
+	  %[1]s --initial=v1.DeploymentConfig --final=apps.openshift.io/v1.DeploymentConfig --confirm`)
+)
+
+func prettyPrintMigrations(versionKinds map[versionKind]versionKind) string {
+	lines := make([]string, 0, len(versionKinds))
+	for initial, final := range versionKinds {
+		line := fmt.Sprintf("		- %s.%s --> %s.%s", initial.APIVersion, initial.Kind, final.APIVersion, final.Kind)
+		lines = append(lines, line)
+	}
+
+	return strings.Join(lines, "\n")
+}
+
+type versionKind struct {
+	APIVersion string
+	Kind       string
+}
+
+func (vk versionKind) GroupVersionKind() (schema.GroupVersionKind, error) {
+	groupVer, err := schema.ParseGroupVersion(vk.APIVersion)
+	if err != nil {
+		return schema.GroupVersionKind{}, err
+	}
+	return groupVer.WithKind(vk.Kind), nil
+}
+
+type MigrateLegacyHPAOptions struct {
+	// maps initial gvks to final gvks in the same format
+	// as HPAs use (CrossVersionObjectReferences) for ease of access.
+	finalVersionKinds map[versionKind]versionKind
+
+	hpaClient autoscalingclient.AutoscalingInterface
+
+	migrate.ResourceOptions
+}
+
+func (o *MigrateLegacyHPAOptions) Bind(c *cobra.Command) {
+	o.ResourceOptions.Bind(c)
```

Comment:
Just call this in `NewCmdMigrateLegacyHPA` and drop the Bind.

Reply:
yep. Used to have more options here -- simplified and forgot to strip it off